standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/Msck.java

```diff
@@ -240,6 +246,76 @@ public int repair(MsckInfo msckInfo) throws TException, MetastoreException, IOEx
           }
         }
 
+        // Generate small files warnings only for partsNotInMs
+        try {
+          // Threshold in bytes for average file size considered "small"
+          final long threshold =
+                  MetastoreConf.getLongVar(getConf(), MetastoreConf.ConfVars.MSCK_SMALLFILES_AVG_SIZE);
+
+          // Collect partition names
+          final List<String> names = new ArrayList<>(partsNotInMs.size());
+          for (CheckResult.PartitionResult pr : partsNotInMs) {
+            final String name = pr.getPartitionName();
+            if (name != null && !name.isEmpty()) {
+              names.add(name);
+            }
+          }
+
+          // Batch over the names to limit single-RPC payload.
+          final int BATCH = 1000;
+          final Map<String, Partition> byName = new HashMap<>(names.size() * 2);
+
+          for (int i = 0; i < names.size(); i += BATCH) {
+            final List<String> batch = names.subList(i, Math.min(i + BATCH, names.size()));
+            final GetPartitionsByNamesRequest req = new GetPartitionsByNamesRequest(table.getDbName(), table.getTableName());
+            req.setNames(batch);
+
+            // In this branch, getPartitionsByNames returns List<Partition>
+            List<Partition> plist;
+            try {
+              @SuppressWarnings("unchecked")
+              List<Partition> tmp = (List<Partition>) getMsc().getPartitionsByNames(req);
+              plist = (tmp != null) ? tmp : Collections.emptyList();
+            } catch (NoSuchObjectException e) {
+              plist = Collections.emptyList();
+            }
+
+            for (Partition p : plist) {
+              final String pName = Warehouse.makePartName(table.getPartitionKeys(), p.getValues());
+              byName.put(pName, p);
+            }
+          }
+
+          // Build small-files stats for partitions that have quick HMS stats.
+          final Map<String, String> smallFilesStats = new TreeMap<>();
+
+          for (String pName : names) {
```

Comment:
why not `for (Partition p : plist)`

Reply:
They are pretty much the same. 
We need to do `smallFilesStats.put(pName, "avgBytes=" + avg + ", partition total files=" + numFiles + ", totalBytes=" + totalSize);` later, so it's easier to use pName to iterate.